elasticx/upsert_response.go

```diff
@@ -0,0 +1,25 @@
+package elasticx
+
+type UpsertResponse[T any] struct {
+	Result UpsertResult `json:"result"`
+	Meta   T            `json:"meta"`
+}
+
+type UpsertResult string
+
+const (
+	UpsertResultCreated UpsertResult = "created"
+	UpsertResultUpdated UpsertResult = "updated"
+)
+
+func (ur UpsertResult) String() string {
+	return string(ur)
+}
+
+func (ur UpsertResult) IsCreated() bool {
```

Comment:
```suggestion
func (ur UpsertResult) Created() bool {
```

To follow the the idiomatic naming convention that favor shorter, simpler names that clearly express intent